### ISSUE TITLE
MC tag clarification

### ISSUE THREAD

**Issue by yfarjoun** [samtools/hts-specs#163]
Hi folks.

We are somewhat confused about the definition of the MC tag in the case of secondary and supplementary alignments. I was hoping to find clarity here.

Consider the following (somewhat contrived, but based on a true story) case:

rec1 primary first pos=100 mate-pos=200 
rec1 primary second pos=200 mate-pos=100
rec1 secondary first pos=100 mate-pos=200 (different CIGAR than the primary)
rec1 secondary second pos=200 maste-pos=100 (different CIGAR than the primary)

While for DNA-seq this is very unlikely, for RNA-seq, this happens since the secondary alignments can be due to alternate alignments of an exon. 

The MC tags of the primaries should always represent the CIGAR string of the other **primary** alignment (I presume), but what about the MC tags of the **secondary** alignments? should the point at the primary alignments, or the secondary? This is not specified one way or the other in the spec. Perhaps MC is undefined in the spec and should only be allowed for primary alignments? 

Same question pertains to Supplementary alignments. Should the MC be "chained"?

**Comment by yfarjoun** [comment]
Given that the definition of PNEXT and RNEXT refers specifically to the primary alignments, I think that it is most consistent that the MC refers to the CIGAR of the primary "mate" (i.e. NEXT) alignment. 

If there are no objections I'll submit a PR to this effect.

**Comment by jkbonfield** [comment]
I think the definitions of PNEXT/RNEXT aren't ideal either, but yes I agree MC, PNEXT and RNEXT should all be considered together and point to the same thing.

Where I find the wording poor on PNEXT is in insisting that it always points to the next primary.  There are situations, eg a read-pair mapping consistently (as a pair) HERE and also consistently as a pair THERE, that it is desirable for a secondary alignment to have PNEXT pointing to its own secondary alignment mate.

Put simply, this comes down to a situation by situation thing and is rather aligner specific in current usage.  Writing that down in a sensible way is challenging.

**Comment by d-cameron** [comment]
>Same question pertains to Supplementary alignments. Should the MC be "chained"?

My code assumes that for secondary, supplementary alignments, MC points to the non-supplementary alignment of the secondary alignment.


That of course assumes secondary alignments are distinct, stand-alone alignments, and the aligner did not interpret the specs as allowing for an alignment graph (see https://sourceforge.net/p/samtools/mailman/message/34489228/ )
